qutip/metrics.py

```diff
@@ -343,7 +343,8 @@ def hellinger_dist(A, B, sparse=False, tol=0):
     return np.sqrt(2.0 * np.maximum(0., (1.0 - np.real(np.sum(eigs)))))
 
 
-def dnorm(A, B=None, solver="CVXOPT", verbose=False, force_solve=False):
+def dnorm(A, B=None, solver="CVXOPT", verbose=False, force_solve=False,
+          dense_memoized_solve=True):
```

Comment:
I have doubts with this, because the tradeoff is not made explicit but I understand that simplicity is better. Shouldn''t sparse be the default behaviour if memoisation does not matter to the end user?

Reply:
The main point here is that the keyword is much too long - nobody wants to type all that, and it's inconsistent with other parts of QuTiP where the dense/sparse keyword toggle is called `sparse`.  The choice of default is a secondary concern - I just wrote `sparse=False` because that matched what you had here.

The memoisation is an internal feature to help keep things fast.  We can memoise the slow part of both the sparse and the dense paths without any problems, so there's no reason to "leak" the implementation through to the user.

Reply:
To be clear, long descriptive names are _usually_ good in internal code.  It's just an issue here because of the inconsistency, and because it includes information that isn't really necessary for the user.